### PR TITLE
Supports Python 3.11

### PR DESCRIPTION
**Pull request by Hazealign**
This PR makes emcache support Python 3.11. (Related Issue: #79)
I check it with GitHub Actions already. ([Commit Check](https://github.com/Hazealign/emcache/actions/runs/3403064648), [Deployment Check](https://github.com/Hazealign/emcache/actions/runs/3403064663)) Only one error is I can't deploy to PyPI in my forked repository. You can check the full logs in link. 

# Changes

 - **I didn't changed any source code.**
 - Update `Cython` to latest version for Support Python 3.11
 - Remove `asynctest` dependency in Unit Tests. `asynctest` is not maintained anymore. And `asynctest` doesn't support Python 3.11. I replaced `AsyncMagicMock` or `CoroutineMock` to `unittest.mock.AsyncMock`. It works well.
 - I fixed some test code. In Python 3.11, Passing coroutine objects to `asyncio.wait()` directly is forbidden.
 - Updated `manylinux2010` to `manylinux2014` dues to `manylinux2010` is EOL, and don't support Python 3.11.

# BREAKING CHANGES

 - `unittest.mock.AsyncMock` is introduced with Python 3.8. So we can't test with Python 3.7 anymore.
 - After this PR, We should discontinue support Python 3.7. Or we should handle only Python 3.7.

### CODE REVIEW

File: bin/build_manylinux1_wheel.sh [hunk 1]
```diff
@@ -24,6 +24,12 @@ then
     PIP=/opt/python/cp310-cp310/bin/pip
     CYTHON=/opt/python/cp310-cp310/bin/cython
     AUDITWHEEL=/opt/python/cp310-cp310/bin/auditwheel
+elif [ $PYTHON_VERSION == "3.11" ]
+then
+    PYTHON=/opt/python/cp311-cp311/bin/python
+    PIP=/opt/python/cp311-cp311/bin/pip
+    CYTHON=/opt/python/cp311-cp311/bin/cython
+    AUDITWHEEL=/opt/python/cp311-cp311/bin/auditwheel
```
Comment (pfreixes) [inline bin/build_manylinux1_wheel.sh:32]: I guess that we can remove this https://github.com/emcache/emcache/pull/86/files#diff-6688e3e7b448091a9e4bb85ef4145d8f1f0f0d7f013ec6dee77300187f4bd80cR3 right?

**Review by pfreixes — CHANGES_REQUESTED** [review]
Nice addition, just one small comment, also please lets make official the new target version by updating the `version.py` [1] file in this PR. Should be 1.0.0.

Also we mostly forget to update the `CHANGELOG` file, if you could add a new entry for the future `1.0.0` version, that would be awesome!

[1] https://github.com/emcache/emcache/blob/master/emcache/version.py#L1

File: bin/build_manylinux1_wheel.sh (hunk 1 above)
Comment (Hazealign) [reply bin/build_manylinux1_wheel.sh:32]: Oh. I think I can remove 3.7 branch. Thanks. :)

**Comment by Hazealign** [comment]
I was too busy on weekdays, so I couldn't make time for this. I'll start to do this. :)

File: bin/build_manylinux1_wheel.sh (hunk 1 above)
Comment (Hazealign) [reply bin/build_manylinux1_wheel.sh:26]: @pfreixes Do you think should we can remove 3.11 branch? I think it'll be necessary.

Comment (pfreixes) [reply bin/build_manylinux1_wheel.sh:26]: Why? arent you adding support for Python 3.11?

Comment (Hazealign) [reply bin/build_manylinux1_wheel.sh:26]: @pfreixes

This process is required to build manylinux wheel. And It's branched with supported Python version.

Before this PR, we supported Python 3.7 to 3.10. After this PR, we supports 3.8 to 3.11. I forgot to remove if python version is 3.7, so I removed it.

Should I need to fix more?

Comment (pfreixes) [reply bin/build_manylinux1_wheel.sh:26]: nops its ok

**Review by pfreixes — APPROVED** [review]

**Pull request merged**